engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -2293,21 +2290,6 @@ private void migrateThroughHypervisorOrStorage(StoragePool destPool, VMInstanceV
         }
     }
 
-    private void preStorageMigrationStateCheck(StoragePool destPool, VMInstanceVO vm) {
-        if (destPool == null) {
-            throw new CloudRuntimeException("Unable to migrate vm: missing destination storage pool");
-        }
-
-        checkDestinationForTags(destPool, vm);
-        try {
-            stateTransitTo(vm, Event.StorageMigrationRequested, null);
-        } catch (final NoTransitionException e) {
-            String msg = String.format("Unable to migrate vm: %s", vm.getUuid());
-            s_logger.debug(msg);
-            throw new CloudRuntimeException(msg, e);
-        }
```

Comment:
@rafaelweingartner @RodrigoDLopez  do we need to remove this part of code as well? This will be changing VM state.
Regarding previously agreed behaviour I'm not completely aware of them. I'll go through older discussions regarding the same. It will be better if others can comment. +1 from me if this is what we already agreed and implemented.